pkg/events/derive/hidden_kernel_module.go

```diff
@@ -25,36 +26,83 @@ import (
 
 var (
 	foundHiddenKernModsCache *lru.Cache
+	wakeupChannel            = make(chan ScanRequest)
```

Comment:
This introduces pretty severe coupling between the big tracee object and the derive module.
Any problem with the channel now like being full and somehow missing the read from it, would cause a lock in the channel and potentially a lock in tracee as a whole.
However I see why this happened, the logic of scanning would, in a regular ebpf event, likely be handled in the `processEvents` part of the pipeline.
I think this shows we need another post-derive process stage in the pipeline to accommodate derived event processing.

Reply:
"Any problem with the channel now like being full and somehow missing the read from it, would cause a lock in the channel and potentially a lock in tracee as a whole."

The receiving end runs in a goroutine so can't block tracee. I could publish to the channel in a goroutine as well, if the thread that runs the derived event code is the same one who does all the other stuff.

Reply:
It's also in its own pipeline thread. But the problem of extendability remains (the code can't be separated out from tracee as a whole).

Reply:
> the logic of scanning would, in a regular ebpf event, likely be handled in the processEvents part of the pipeline.

> I think this shows we need another post-derive process stage in the pipeline to accommodate derived event processing.

I agree with @NDStrahilevitz here. We need a `processEvents` for the derived events. I would say this "wakeupChannel" should be global - for the derived pkgs - for all derived events scheduling callbacks to it, and not particular to a single event derived logic implementation. The callback would have a pointer for a function to be executed, with arguments (just like any other work queue implementation).

BUUUT I also think its too much for this PR. I think @OriGlassman should finish this work, and we should stop with all the uprobe triggering stuff until we DECOUPLE the uprobe/hooking functions/logic from the ebpf package. I prefer the opportunistic refactor approach but it did not happen since the first hooking logic was added, so...

Also, please note some things that CAN be done now:

1. spin off the hidden_kernel_module related functions to a new file in the golang ebpf package
2. spin off all the eBPF functions related to hidden kernel modules to a eBPF source file
3. improve overall documentation (not only function descriptions and summaries, but also the full logic).

About (3):

Logic to be documented for first code readers: from time to time trigger the hidden module investigation, multiple techniques using tailcalls, sending to userland but not reporting, investigating again, sending back and forth until finally user is warned.

Note: I know there is some description to this already, but it does not correspond to the entire logic and chronology of what happens when. Also, the code is already open source, so by fully documenting it we are only helping ourselves to maintain it in a near future (and not others, that will INEVITABLY copy the logic like done before).